lib/ovirt_metrics/models/ovirt_history.rb

```diff
@@ -1,8 +1,17 @@
 module OvirtMetrics
   class OvirtHistory < ActiveRecord::Base
+    attr_writer :connection_specification_name if ActiveRecord::VERSION::MAJOR < 5
+
     self.abstract_class = true
     self.pluralize_table_names = false
 
+    def self.connection_specification_name
+      if !defined?(@connection_specification_name) || @connection_specification_name.nil?
+        return self == OvirtHistory ? OvirtMetrics.config.connection_specification_name : superclass.connection_specification_name
+      end
```

Comment:
I don't know of a better way though, so I'm fine with taking the conditionals as is since we had some issues without these conditionals as they are.